### PR TITLE
Feature/update customer courier

### PR DESCRIPTION
**Pull request by jake6642**
This fixes the auto-increment for customers/couriers and allows the user to update the customer or courier. It also now saves imports to the databse wheras it didn't before. 
You can also now activate/deactivate couriers and customers.

### CODE REVIEW

File: src/acme/pd/Courier.java [hunk 1]
```diff
@@ -27,14 +30,18 @@ public void setCourierNumber(int courierNumber) {
     }
 
     public static int getNextCourierNumber() {
-		/*try {
-			//FIXME
-			Courier highest = PersistableEntity.querySingle(Courier.class, "select c from COURIER c ORDER BY NUMBER DESC", Collections.EMPTY_MAP);
-			return highest.getCourierNumber() + 1;
+		try {
```
Comment (jake6642) [inline src/acme/pd/Courier.java]: I know it isn't the cleanest, but it gives us the auto-increment that we weren't getting before.

File: src/acme/ui/AcmeUI.java [hunk 2]
```diff
@@ -137,96 +131,96 @@ public void setPanel(AcmeBaseJPanel panel) {
...
+        CustomerListUI customerListUI = new CustomerListUI();
+        this.setPanel(customerListUI);
     }
 
-    // Everyone will tie in their panel like this.  Replace my example with your code
+    // Add or update a customer
     public void customerAddUpdate(Customer customer, Consumer<Customer> onSaveFunc) {
         // This will be called with null to create
-        CustomerCreatePanel panel = new CustomerCreatePanel(onSaveFunc);
+        CustomerCreatePanel panel = new CustomerCreatePanel(customer, onSaveFunc);
```
Comment (jake6642) [inline src/acme/ui/AcmeUI.java]: This now takes a customer as a parameter so we can update if necessary.

File: src/acme/ui/CustomerListUI.java [hunk 3]
```diff
@@ -155,14 +153,15 @@ public static void main(String[] args) {
...
+        //customerName = customerName;
         Customer customerToPass = null;
         for (int x = 0; x < this.customersArray.length; x++) {
-            if (this.customersArray[x].getName().equals(customerName)) {
+            if (this.customersArray[x].getName().equals(customerName)) {                
                 customerToPass = this.customersArray[x];
             }
         }
-        this.getAcmeUI().customerAddUpdate(customerToPass, c -> this.getAcmeUI().customerList());
+        AcmeUI acme = this.getAcmeUI();
```
Comment (jake6642) [inline src/acme/ui/CustomerListUI.java]: This fixed a stinkin' null pointer problem that took forever to nail down...

Comment (epielemeier) [reply src/acme/ui/CustomerListUI.java]: Yep, thanks Java garbage collection!

**Pull request merged**